src/uct/ib/base/ib_iface.c

```diff
@@ -244,61 +248,29 @@ int uct_ib_iface_is_reachable(const uct_iface_h tl_iface, const uct_device_addr_
     }
 }
 
-void uct_ib_iface_fill_ah_attr(uct_ib_iface_t *iface, const uct_ib_address_t *ib_addr,
-                               uint8_t path_bits, struct ibv_ah_attr *ah_attr)
-{
-    uint8_t is_global;
-
-    memset(ah_attr, 0, sizeof(*ah_attr));
-
-    uct_ib_address_unpack(ib_addr, &ah_attr->dlid, &is_global, &ah_attr->grh.dgid);
-    ah_attr->sl            = iface->config.sl;
-    ah_attr->src_path_bits = path_bits;
-    ah_attr->dlid         |= path_bits;
-    ah_attr->port_num      = iface->config.port_num;
-
-    /* Create a global address only if we cannot reach the destination using local
-     * address. It means either an Ethernet address, or IB address on a different subnet.
-     */
-    if (is_global &&
-        ((iface->addr_type == UCT_IB_ADDRESS_TYPE_ETH) ||
-         (iface->addr_type == UCT_IB_ADDRESS_TYPE_GLOBAL) ||
-         (iface->gid.global.subnet_prefix != ah_attr->grh.dgid.global.subnet_prefix)))
-    {
-        ah_attr->is_global      = 1;
-        ah_attr->grh.sgid_index = iface->config.gid_index;
-    } else {
-        ah_attr->is_global      = 0;
-    }
-}
-
 ucs_status_t uct_ib_iface_create_ah(uct_ib_iface_t *iface,
-                                    const uct_ib_address_t *ib_addr,
-                                    uint8_t path_bits,
-                                    struct ibv_ah **ah_p,
-                                    int *is_global_p)
+                                    struct ibv_ah_attr *ah_attr,
```

Comment:
better make it const*

Reply:
will not work: ibv_create_ah call has parameters:
```
(struct ibv_pd *pd, struct ibv_ah_attr *attr)
```
without const specification

Reply:
right :(